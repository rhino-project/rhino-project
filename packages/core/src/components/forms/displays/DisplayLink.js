import { useMemo } from 'react';
import { useController } from 'react-hook-form';

const DisplayLink = ({ accessor, children, empty = '-', ...props }) => {
  const { path } = props;
  const {
    field: { value: fieldValue }
  } = useController({ name: path });

  const value = useMemo(() => (accessor ? accessor(fieldValue) : fieldValue), [
    accessor,
    fieldValue
  ]);

  return <div>{value ? <a href={value}>{children || value}</a> : empty}</div>;
};

export default DisplayLink;
