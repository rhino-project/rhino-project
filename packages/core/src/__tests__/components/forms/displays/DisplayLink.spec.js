import { sharedDisplayTests } from './sharedDisplayTests';
import DisplayLink from '../../../../components/forms/displays/DisplayLink';

describe('DisplayLink', () => {
  sharedDisplayTests(DisplayLink);
});
