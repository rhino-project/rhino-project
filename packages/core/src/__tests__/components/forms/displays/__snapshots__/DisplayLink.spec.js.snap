// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`DisplayLink > renders inside of model context 1`] = `
<DocumentFragment>
  <div>
    -
  </div>
</DocumentFragment>
`;

exports[`DisplayLink > should render with global override shorthand 1`] = `
<DocumentFragment>
  <div>
    Bar
  </div>
</DocumentFragment>
`;
